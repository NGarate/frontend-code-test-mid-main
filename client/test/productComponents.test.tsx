import React from 'react';
import renderer from 'react-test-renderer';
import ProductDescription from '../components/product/description';
import ProductImage from '../components/product/image';
import ProductMain from '../components/product/image';
import ProductSpecs from '../components/product/specs';

test('renders the ProductDescription correctly', () => {
  const tree = renderer.create(<ProductDescription />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders the ProductImage correctly', () => {
  const tree = renderer.create(<ProductImage />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders the ProductMain correctly', () => {
  const tree = renderer.create(<ProductMain />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders the ProductSpecs correctly', () => {
  const tree = renderer.create(<ProductSpecs />).toJSON();
  expect(tree).toMatchSnapshot();
});
