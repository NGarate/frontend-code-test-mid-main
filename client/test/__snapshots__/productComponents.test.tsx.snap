// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the ProductDescription correctly 1`] = `
<div
  className="rt-Box sc-6ba452cb-0 hPTzLy"
>
  <div
    className="sc-6ba452cb-1 fIpZXk"
  >
    <h2
      className="rt-Heading rt-r-size-6"
    >
      Description
    </h2>
    <div
      className="rt-Box rt-r-pt-4"
    />
  </div>
</div>
`;

exports[`renders the ProductImage correctly 1`] = `
<div
  data-radix-aspect-ratio-wrapper=""
  style={
    Object {
      "paddingBottom": "100%",
      "position": "relative",
      "width": "100%",
    }
  }
>
  <div
    className="sc-bd8b558e-0 kkXGrV"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <img
      alt="Phillips Plumen"
      className="sc-bd8b558e-1 gAJRKH"
      src="/philips-plumen.jpg"
    />
  </div>
</div>
`;

exports[`renders the ProductMain correctly 1`] = `
<div
  data-radix-aspect-ratio-wrapper=""
  style={
    Object {
      "paddingBottom": "100%",
      "position": "relative",
      "width": "100%",
    }
  }
>
  <div
    className="sc-bd8b558e-0 kkXGrV"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <img
      alt="Phillips Plumen"
      className="sc-bd8b558e-1 gAJRKH"
      src="/philips-plumen.jpg"
    />
  </div>
</div>
`;

exports[`renders the ProductSpecs correctly 1`] = `
<div
  className="rt-Box sc-ae3d4568-0 iXHXzu"
>
  <h2
    className="rt-Heading rt-r-size-6"
  >
    Specifications
  </h2>
  <dl
    className="rt-Text rt-DataListRoot rt-r-orientation-horizontal rt-r-size-2 sc-ae3d4568-1 iMxreY"
    style={Object {}}
  >
    <div
      className="rt-DataListItem"
    >
      <dt
        className="rt-DataListLabel rt-r-min-w sc-ae3d4568-2 jYgHMH"
        style={
          Object {
            "--min-width": "88px",
          }
        }
      >
        Brand
      </dt>
      <dd
        className="rt-DataListValue"
      />
    </div>
    <div
      className="rt-DataListItem"
    >
      <dt
        className="rt-DataListLabel rt-r-min-w sc-ae3d4568-2 jYgHMH"
        style={
          Object {
            "--min-width": "88px",
          }
        }
      >
        Item weight (g)
      </dt>
      <dd
        className="rt-DataListValue"
      />
    </div>
    <div
      className="rt-DataListItem"
    >
      <dt
        className="rt-DataListLabel rt-r-min-w sc-ae3d4568-2 jYgHMH"
        style={
          Object {
            "--min-width": "88px",
          }
        }
      >
        Dimensions (cm)
      </dt>
      <dd
        className="rt-DataListValue"
      >
        undefined x undefined x undefined
      </dd>
    </div>
    <div
      className="rt-DataListItem"
    >
      <dt
        className="rt-DataListLabel rt-r-min-w sc-ae3d4568-2 jYgHMH"
        style={
          Object {
            "--min-width": "88px",
          }
        }
      >
        Item Model number
      </dt>
      <dd
        className="rt-DataListValue"
      />
    </div>
    <div
      className="rt-DataListItem"
    >
      <dt
        className="rt-DataListLabel rt-r-min-w sc-ae3d4568-2 jYgHMH"
        style={
          Object {
            "--min-width": "88px",
          }
        }
      >
        Colour
      </dt>
      <dd
        className="rt-DataListValue"
      />
    </div>
  </dl>
</div>
`;
